src/main/java/com/wojciechbarwinski/demo/epic_board_games_shop/services/OrderService.java

```diff
@@ -46,4 +48,53 @@ public List<OrderResponseDTO> getAllOrdersBySearchingData(OrderSearchRequestDTO
                 .map(mapper::mapOrderToOrderResponseDTO)
                 .toList();
     }
+
+    public OrderResponseDTO proceedOrderAfterConfirm(Long id) {
+        Order order = orderRepository.save(updateOrderStatus(id, OrderStatus.CONFIRMED));
+
+        //here will be method that create payment-link and sent it to client mail to pay OR to decline order
+        return mapper.mapOrderToOrderResponseDTO(order);
+    }
+
+    public void rejectOrder(Long id) {
+        Order order = updateOrderStatus(id, OrderStatus.CANCELLED);
+        orderRepository.save(order);
+    }
+
+    public void updateStatusFromWarehouse(OrderDataFromWarehouseDTO orderDataDTO) {
+        Order order = updateOrderStatus(orderDataDTO.orderId(), orderDataDTO.status());
+        orderRepository.save(order);
+    }
+
+    public void cancelOrder(Long hashId) {
+        Long id = hashId;// method to read hashId from link as order ID
+        Order order = updateOrderStatus(id, OrderStatus.CANCELLED);
+        //method to start back-cash process
+        orderRepository.save(order);
+    }
+
+    public void proceedOrderAfterPayment(Long hashId) {
+        Long id = hashId;// method to read hashId from link as order ID
+        Order order = updateOrderStatus(id, OrderStatus.PAID); //tmp useless
+
+        OrderDataToWarehouseDTO orderDataToWarehouseDTO = mapper.mapOrderToOrderDataToWarehouse(order);
+        OrderDataFromWarehouseDTO orderDataFromWarehouseDTO = client.sendOrderToWarehouse(orderDataToWarehouseDTO);
+
+        order.setOrderStatus(orderDataFromWarehouseDTO.status());
+
+        orderRepository.save(order);
+    }
+
+    private Order getOrder(Long id) {
+        return orderRepository.findById(id)
+                .orElseThrow(() -> new OrderNotFoundException(id));
+    }
+
+    private Order updateOrderStatus(Long id, OrderStatus status) {
+        Order order = getOrder(id);
+        //TODO moj pomysł na validacje przejsc statusów to w metodzie .setOrderStatus umieścić statyczną metodę validującą po prostu
```

Comment:
To już przestałby być pojo... lepiej zrobić to w osobnej klasie i wtedy można napisać do tego prawilny test jednostkowy (a nie test encji). Ogólnie DDD dopuszcza takie podejście, a nawet uważa je za prawilniejsze, ale na projekcie wzbudziłbyś tym sporo dyskusji :P

Reply:
Dobra ale to samo w sobie bedzie jako osobne zadanie. Ja ten serwis podziele i zrobie miejsce na tą walidacje ale ją samą zrobie w osobnym zadaniu